### DISCUSSION TITLE
About new pagination feature

### DISCUSSION THREAD

**Discussion by canob** [grafana/grafana-infinity-datasource#680] [Q&A]
Hi everyone,

I managed to connect OpenObserve to Grafana with Infinity Plugin, with JSON Backend POST, using this URL, http://openobserve:5080/api/default/_search?type=logs, and with this Raw body type with JSON Content Type:
{"query":{"sql":"select * from \"windows\"","start_time":${__from}000,"end_time":${__to}000,"from":0,"size":10000}}

I want to "manage" that "size" field with new pagination feature. Is it possible?
![imagen](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/30271927/00f5bc68-4d93-4c3e-aefd-328a8942d887)

Thanks in advance for your help.

**Comment by canob** [comment]
Nobody?

**Comment by lakshay2711** [comment]
Hello @canob ,

did you got any solution?
Also I have a kind of similar question regarding the **size** have you ever tried to provide a size greater than 10000 if your ES has more than 10K records? Does it work?
For me it gives a 400 bad request error in grafana. Does Grafana has a limitation or are you aware of any workaround for this issue ?

**Reply by canob** [reply]
Hi @lakshay2711.
No, I gave up, and I didn't get any response.
I'm using OpenObserve, but in the case of ElasticSearch, I think that is more complicated, because you need to use an additional parameter, search_after, https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html, and I don't know how you can manage that kind of things with this new pagination feature of Infinity Datasource Plugin.
